# 판다스 문제2

In [1]:
import pandas as pd

### 1. 주어진 csv 파일을 업로드하고 변수를 사용하여 csv 파일을 판다스 함수를 이용하여 읽어들이시오.

In [ ]:
# 파일을 업로드하시오.
from google.colab import files
uploaded = files.upload() # 파일 선택 창을 통해 파일을 불러와 'upload'라는 변수로 파일 지정

In [3]:
# csv 파일을 변수로 받아들이고 출력하시오.
heartDiseaseData = pd.read_csv('heart_disease_uci.csv')
heartDiseaseData

id  age     sex        dataset               cp  trestbps   chol    fbs  \
0      1   63    Male      Cleveland   typical angina     145.0  233.0   True   
1      2   67    Male      Cleveland     asymptomatic     160.0  286.0  False   
2      3   67    Male      Cleveland     asymptomatic     120.0  229.0  False   
3      4   37    Male      Cleveland      non-anginal     130.0  250.0  False   
4      5   41  Female      Cleveland  atypical angina     130.0  204.0  False   
..   ...  ...     ...            ...              ...       ...    ...    ...   
915  916   54  Female  VA Long Beach     asymptomatic     127.0  333.0   True   
916  917   62    Male  VA Long Beach   typical angina       NaN  139.0  False   
917  918   55    Male  VA Long Beach     asymptomatic     122.0  223.0   True   
918  919   58    Male  VA Long Beach     asymptomatic       NaN  385.0   True   
919  920   62    Male  VA Long Beach  atypical angina     120.0  254.0  False   

              restecg  thalch  exang  oldpeak        slope   ca  \
0      lv hypertrophy   150.0  False      2.3  downsloping  0.0   
1      lv hypertrophy   108.0   True      1.5         flat  3.0   
2      lv hypertrophy   129.0   True      2.6         flat  2.0   
3              normal   187.0  False      3.5  downsloping  0.0   
4      lv hypertrophy   172.0  False      1.4    upsloping  0.0   
..                ...     ...    ...      ...          ...  ...   
915  st-t abnormality   154.0  False      0.0          NaN  NaN   
916  st-t abnormality     NaN    NaN      NaN          NaN  NaN   
917  st-t abnormality   100.0  False      0.0          NaN  NaN   
918    lv hypertrophy     NaN    NaN      NaN          NaN  NaN   
919    lv hypertrophy    93.0   True      0.0          NaN  NaN   

                  thal  num  
0         fixed defect    0  
1               normal    2  
2    reversable defect    1  
3               normal    0  
4               normal    0  
..                 ...  ...  
915                NaN    1  
916                NaN    0  
917       fixed defect    2  
918                NaN    0  
919                NaN    1  

[920 rows x 16 columns]

### 2. 판다스의 속성 및 함수를 이용하여 다음 문제들을 푸시오.

#### 2-1.DataFrame의 행과 열 크기를 출력하시오.

In [4]:
heartDiseaseData.shape

(920, 16)

#### 2-2.DataFrame의 총 데이터 건수와 각 칼럼별 데이터 타입을 출력하시오.

In [5]:
heartDiseaseData.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 920 entries, 0 to 919
Data columns (total 16 columns):
 #   Column    Non-Null Count  Dtype  
---  ------    --------------  -----  
 0   id        920 non-null    int64  
 1   age       920 non-null    int64  
 2   sex       920 non-null    object 
 3   dataset   920 non-null    object 
 4   cp        920 non-null    object 
 5   trestbps  861 non-null    float64
 6   chol      890 non-null    float64
 7   fbs       830 non-null    object 
 8   restecg   918 non-null    object 
 9   thalch    865 non-null    float64
 10  exang     865 non-null    object 
 11  oldpeak   858 non-null    float64
 12  slope     611 non-null    object 
 13  ca        309 non-null    float64
 14  thal      434 non-null    object 
 15  num       920 non-null    int64  
dtypes: float64(5), int64(3), object(8)
memory usage: 115.1+ KB


#### 2-3.DataFrame의 각 칼럼별 숫자형 데이터의 통계들을 출력하시오.

In [6]:
heartDiseaseData.describe()

id         age    trestbps        chol      thalch     oldpeak  \
count  920.000000  920.000000  861.000000  890.000000  865.000000  858.000000   
mean   460.500000   53.510870  132.132404  199.130337  137.545665    0.878788   
std    265.725422    9.424685   19.066070  110.780810   25.926276    1.091226   
min      1.000000   28.000000    0.000000    0.000000   60.000000   -2.600000   
25%    230.750000   47.000000  120.000000  175.000000  120.000000    0.000000   
50%    460.500000   54.000000  130.000000  223.000000  140.000000    0.500000   
75%    690.250000   60.000000  140.000000  268.000000  157.000000    1.500000   
max    920.000000   77.000000  200.000000  603.000000  202.000000    6.200000   

               ca         num  
count  309.000000  920.000000  
mean     0.676375    0.995652  
std      0.935653    1.142693  
min      0.000000    0.000000  
25%      0.000000    0.000000  
50%      0.000000    1.000000  
75%      1.000000    2.000000  
max      3.000000    4.000000

In [7]:
heartDiseaseData.columns

Index(['id', 'age', 'sex', 'dataset', 'cp', 'trestbps', 'chol', 'fbs',
       'restecg', 'thalch', 'exang', 'oldpeak', 'slope', 'ca', 'thal', 'num'],
      dtype='object')

### 3. 다음 표를 참고하여 칼럼명을 한글로 바꾸시오.
| 칼럼명 | 한글 |
| :--- | :--- |
| id | 식별자 |
| age | 나이 |
| sex | 성별 |
| cp | 가슴 통증 |
| trestbps | 혈압 |
| chol | 콜레스테롤 |
| fbs | 공복 혈당 |
| restecg | 심전도 |
| thalach | 최대 심박수 |
| exang | 협심증 |
| oldpeak | ST 하강 |
| slope | ST 기울기 |
| ca | 주요 혈관 수 |
| thal | 탈라세미아 |
| num | 심장병 진단 |

In [8]:
column_rename_dict = {
    'id': '식별자',
    'age': '나이',
    'sex': '성별',
    'cp': '가슴 통증',
    'trestbps': '혈압',
    'chol': '콜레스테롤',
    'fbs': '공복 혈당',
    'restecg': '심전도',
    'thalch': '최대 심박수',
    'exang': '협심증',
    'oldpeak': 'ST 하강',
    'slope': 'ST 기울기',
    'ca': '주요 혈관 수',
    'thal': '탈라세미아',
    'num': '심장병 진단'
}
heartDiseaseData.rename(columns=column_rename_dict, inplace=True)
heartDiseaseData

식별자  나이      성별        dataset            가슴 통증     혈압  콜레스테롤  공복 혈당  \
0      1  63    Male      Cleveland   typical angina  145.0  233.0   True   
1      2  67    Male      Cleveland     asymptomatic  160.0  286.0  False   
2      3  67    Male      Cleveland     asymptomatic  120.0  229.0  False   
3      4  37    Male      Cleveland      non-anginal  130.0  250.0  False   
4      5  41  Female      Cleveland  atypical angina  130.0  204.0  False   
..   ...  ..     ...            ...              ...    ...    ...    ...   
915  916  54  Female  VA Long Beach     asymptomatic  127.0  333.0   True   
916  917  62    Male  VA Long Beach   typical angina    NaN  139.0  False   
917  918  55    Male  VA Long Beach     asymptomatic  122.0  223.0   True   
918  919  58    Male  VA Long Beach     asymptomatic    NaN  385.0   True   
919  920  62    Male  VA Long Beach  atypical angina  120.0  254.0  False   

                  심전도  최대 심박수    협심증  ST 하강       ST 기울기  주요 혈관 수  \
0      lv hypertrophy   150.0  False    2.3  downsloping      0.0   
1      lv hypertrophy   108.0   True    1.5         flat      3.0   
2      lv hypertrophy   129.0   True    2.6         flat      2.0   
3              normal   187.0  False    3.5  downsloping      0.0   
4      lv hypertrophy   172.0  False    1.4    upsloping      0.0   
..                ...     ...    ...    ...          ...      ...   
915  st-t abnormality   154.0  False    0.0          NaN      NaN   
916  st-t abnormality     NaN    NaN    NaN          NaN      NaN   
917  st-t abnormality   100.0  False    0.0          NaN      NaN   
918    lv hypertrophy     NaN    NaN    NaN          NaN      NaN   
919    lv hypertrophy    93.0   True    0.0          NaN      NaN   

                 탈라세미아  심장병 진단  
0         fixed defect       0  
1               normal       2  
2    reversable defect       1  
3               normal       0  
4               normal       0  
..                 ...     ...  
915                NaN       1  
916                NaN       0  
917       fixed defect       2  
918                NaN       0  
919                NaN       1  

[920 rows x 16 columns]

### 4. 가슴통증 종류를 출력하시오.

In [9]:
heartDiseaseData['가슴 통증'].value_counts()

가슴 통증
asymptomatic       496
non-anginal        204
atypical angina    174
typical angina      46
Name: count, dtype: int64

### 5. 불린 인덱싱을 이용하여 다음 문제들을 푸시오.

In [10]:
heartDiseaseData

식별자  나이      성별        dataset            가슴 통증     혈압  콜레스테롤  공복 혈당  \
0      1  63    Male      Cleveland   typical angina  145.0  233.0   True   
1      2  67    Male      Cleveland     asymptomatic  160.0  286.0  False   
2      3  67    Male      Cleveland     asymptomatic  120.0  229.0  False   
3      4  37    Male      Cleveland      non-anginal  130.0  250.0  False   
4      5  41  Female      Cleveland  atypical angina  130.0  204.0  False   
..   ...  ..     ...            ...              ...    ...    ...    ...   
915  916  54  Female  VA Long Beach     asymptomatic  127.0  333.0   True   
916  917  62    Male  VA Long Beach   typical angina    NaN  139.0  False   
917  918  55    Male  VA Long Beach     asymptomatic  122.0  223.0   True   
918  919  58    Male  VA Long Beach     asymptomatic    NaN  385.0   True   
919  920  62    Male  VA Long Beach  atypical angina  120.0  254.0  False   

                  심전도  최대 심박수    협심증  ST 하강       ST 기울기  주요 혈관 수  \
0      lv hypertrophy   150.0  False    2.3  downsloping      0.0   
1      lv hypertrophy   108.0   True    1.5         flat      3.0   
2      lv hypertrophy   129.0   True    2.6         flat      2.0   
3              normal   187.0  False    3.5  downsloping      0.0   
4      lv hypertrophy   172.0  False    1.4    upsloping      0.0   
..                ...     ...    ...    ...          ...      ...   
915  st-t abnormality   154.0  False    0.0          NaN      NaN   
916  st-t abnormality     NaN    NaN    NaN          NaN      NaN   
917  st-t abnormality   100.0  False    0.0          NaN      NaN   
918    lv hypertrophy     NaN    NaN    NaN          NaN      NaN   
919    lv hypertrophy    93.0   True    0.0          NaN      NaN   

                 탈라세미아  심장병 진단  
0         fixed defect       0  
1               normal       2  
2    reversable defect       1  
3               normal       0  
4               normal       0  
..                 ...     ...  
915                NaN       1  
916                NaN       0  
917       fixed defect       2  
918                NaN       0  
919                NaN       1  

[920 rows x 16 columns]

#### 5-1. 심전도가 'normal'인 데이터만 추출하여 출력하시오.

In [12]:
heartDiseaseData[heartDiseaseData['심전도'] == 'normal'].head()

식별자  나이      성별    dataset            가슴 통증     혈압  콜레스테롤  공복 혈당     심전도  \
3     4  37    Male  Cleveland      non-anginal  130.0  250.0  False  normal   
5     6  56    Male  Cleveland  atypical angina  120.0  236.0  False  normal   
7     8  57  Female  Cleveland     asymptomatic  120.0  354.0  False  normal   
10   11  57    Male  Cleveland     asymptomatic  140.0  192.0  False  normal   
13   14  44    Male  Cleveland  atypical angina  120.0  263.0  False  normal   

    최대 심박수    협심증  ST 하강       ST 기울기  주요 혈관 수              탈라세미아  심장병 진단  
3    187.0  False    3.5  downsloping      0.0             normal       0  
5    178.0  False    0.8    upsloping      0.0             normal       0  
7    163.0   True    0.6    upsloping      0.0             normal       0  
10   148.0  False    0.4         flat      0.0       fixed defect       0  
13   173.0  False    0.0    upsloping      0.0  reversable defect       0

#### 5-2.나이가 60세 이상인 환자의 성별과 나이를 출력하시오.


In [13]:
heartDiseaseData[heartDiseaseData['나이'] >= 60][['성별', '나이']].head()

성별  나이
0    Male  63
1    Male  67
2    Male  67
6  Female  62
8    Male  63

#### 5-3. 나이가 60세 이상이고 성별이 여성이며 혈압이 130 이상인 환자들을 두 가지 방식으로 추출하여 출력하시오.

In [ ]:
# [] 안에 모든 조건을 집어 넣기
heartDiseaseData[(heartDiseaseData['나이'] >= 60) & (heartDiseaseData['성별'] == 'Female')
                 & (heartDiseaseData['혈압'] >= 130)].head()

식별자  나이      성별    dataset            가슴 통증     혈압  콜레스테롤  공복 혈당  \
6     7  62  Female  Cleveland     asymptomatic  140.0  268.0  False   
27   28  66  Female  Cleveland   typical angina  150.0  226.0  False   
30   31  69  Female  Cleveland   typical angina  140.0  239.0  False   
40   41  65  Female  Cleveland     asymptomatic  150.0  225.0  False   
42   43  71  Female  Cleveland  atypical angina  160.0  302.0  False   

               심전도  최대 심박수    협심증  ST 하강       ST 기울기  주요 혈관 수  \
6   lv hypertrophy   160.0  False    3.6  downsloping      2.0   
27          normal   114.0  False    2.6  downsloping      0.0   
30          normal   151.0  False    1.8    upsloping      2.0   
40  lv hypertrophy   114.0  False    1.0         flat      3.0   
42          normal   162.0  False    0.4    upsloping      2.0   

                탈라세미아  심장병 진단  
6              normal       3  
27             normal       0  
30             normal       0  
40  reversable defect       4  
42             normal       0

In [17]:
# 개별 조건을 변수에 할당하고 이들 변수를 결합하기
cond1 = heartDiseaseData['나이'] >= 60
cond2 = heartDiseaseData['성별'] == 'Female'
cond3 = heartDiseaseData['혈압'] >= 130
heartDiseaseData[cond1 & cond2 & cond3].head()

식별자  나이      성별    dataset            가슴 통증     혈압  콜레스테롤  공복 혈당  \
6     7  62  Female  Cleveland     asymptomatic  140.0  268.0  False   
27   28  66  Female  Cleveland   typical angina  150.0  226.0  False   
30   31  69  Female  Cleveland   typical angina  140.0  239.0  False   
40   41  65  Female  Cleveland     asymptomatic  150.0  225.0  False   
42   43  71  Female  Cleveland  atypical angina  160.0  302.0  False   

               심전도  최대 심박수    협심증  ST 하강       ST 기울기  주요 혈관 수  \
6   lv hypertrophy   160.0  False    3.6  downsloping      2.0   
27          normal   114.0  False    2.6  downsloping      0.0   
30          normal   151.0  False    1.8    upsloping      2.0   
40  lv hypertrophy   114.0  False    1.0         flat      3.0   
42          normal   162.0  False    0.4    upsloping      2.0   

                탈라세미아  심장병 진단  
6              normal       3  
27             normal       0  
30             normal       0  
40  reversable defect       4  
42             normal       0

### 6. 가슴 통증 이름을 기준으로 DataFrame을 오름차순 정렬시키시오.

In [19]:
heartDiseaseData_sorted = heartDiseaseData.sort_values(by = ['가슴 통증'])
heartDiseaseData_sorted

식별자  나이    성별        dataset           가슴 통증     혈압  콜레스테롤  공복 혈당  \
731  732  59  Male  VA Long Beach    asymptomatic  140.0    0.0  False   
651  652  55  Male    Switzerland    asymptomatic  140.0    0.0  False   
655  656  56  Male    Switzerland    asymptomatic  115.0    0.0    NaN   
656  657  56  Male    Switzerland    asymptomatic  120.0    0.0  False   
657  658  56  Male    Switzerland    asymptomatic  120.0    0.0  False   
..   ...  ..   ...            ...             ...    ...    ...    ...   
196  197  69  Male      Cleveland  typical angina  160.0  234.0   True   
199  200  59  Male      Cleveland  typical angina  160.0  273.0  False   
211  212  38  Male      Cleveland  typical angina  120.0  231.0  False   
796  797  49  Male  VA Long Beach  typical angina  130.0    0.0  False   
0      1  63  Male      Cleveland  typical angina  145.0  233.0   True   

                  심전도  최대 심박수    협심증  ST 하강       ST 기울기  주요 혈관 수  \
731  st-t abnormality   117.0   True    1.0         flat      NaN   
651            normal    83.0  False    0.0         flat      NaN   
655  st-t abnormality    82.0  False   -1.0    upsloping      NaN   
656  st-t abnormality   100.0   True   -1.0  downsloping      NaN   
657  st-t abnormality   148.0  False    0.0         flat      NaN   
..                ...     ...    ...    ...          ...      ...   
196    lv hypertrophy   131.0  False    0.1         flat      1.0   
199    lv hypertrophy   125.0  False    0.0    upsloping      0.0   
211            normal   182.0   True    3.8         flat      0.0   
796  st-t abnormality   145.0  False    3.0         flat      NaN   
0      lv hypertrophy   150.0  False    2.3  downsloping      0.0   

                 탈라세미아  심장병 진단  
731                NaN       1  
651  reversable defect       2  
655                NaN       1  
656  reversable defect       2  
657                NaN       2  
..                 ...     ...  
196             normal       0  
199             normal       1  
211  reversable defect       4  
796                NaN       2  
0         fixed defect       0  

[920 rows x 16 columns]

### 7. groupby() 함수를 활용하여 다음 문제들을 푸시오.

#### 7-1. 가슴통증별로 각 칼럼별 데이터 개수를 출력하시오.

In [20]:
heartDiseaseData.groupby('가슴 통증').count()

식별자   나이   성별  dataset   혈압  콜레스테롤  공복 혈당  심전도  최대 심박수  협심증  \
가슴 통증                                                                          
asymptomatic     496  496  496      496  467    484    431  495     470  470   
atypical angina  174  174  174      174  167    164    168  174     167  167   
non-anginal      204  204  204      204  185    196    188  204     186  186   
typical angina    46   46   46       46   42     46     43   45      42   42   

                 ST 하강  ST 기울기  주요 혈관 수  탈라세미아  심장병 진단  
가슴 통증                                                   
asymptomatic       465     377      150    239     496  
atypical angina    167      75       50     57     174  
non-anginal        185     127       85    112     204  
typical angina      41      32       24     26      46

#### 7-2. 성별별로 혈압, 콜레스테롤의 평균을 출력하시오.

In [21]:
heartDiseaseData.groupby('성별')[['혈압', '콜레스테롤']].mean()

혈압       콜레스테롤
성별                            
Female  132.093750  241.481283
Male    132.143498  187.864865

#### 7-3.가슴통증별로 그룹을 나누고 최대 나이값과 평균 최대 심박수 그리고 최소 혈압을 출력하시오.

In [22]:
agg_format = {'나이' : 'max', '최대 심박수' : 'mean', '혈압' : 'min'}
heartDiseaseData.groupby('가슴 통증').agg(agg_format)

나이      최대 심박수    혈압
가슴 통증                                
asymptomatic     77  128.914894  80.0
atypical angina  74  150.856287  98.0
non-anginal      76  144.698925   0.0
typical angina   74  149.523810  95.0

### 8. 결손 데이터를 처리하기 위해 다음 문제들을 푸시오.

#### 8-1. 각 칼럼별로 결손 데이터 개수를 출력하시오.

In [23]:
heartDiseaseData.isna().sum()

식별자          0
나이           0
성별           0
dataset      0
가슴 통증        0
혈압          59
콜레스테롤       30
공복 혈당       90
심전도          2
최대 심박수      55
협심증         55
ST 하강       62
ST 기울기     309
주요 혈관 수    611
탈라세미아      486
심장병 진단       0
dtype: int64

#### 8-2. 혈압, 콜레스테롤, 최대 심박수 칼럼들에 있는 null 값들을 다 평균값으로 번경하시오.

In [24]:
heartDiseaseData['혈압'] = heartDiseaseData['혈압'].fillna(heartDiseaseData['혈압'].mean())
heartDiseaseData['콜레스테롤'] = heartDiseaseData['콜레스테롤'].fillna(heartDiseaseData['콜레스테롤'].mean())
heartDiseaseData['최대 심박수'] = heartDiseaseData['최대 심박수'].fillna(heartDiseaseData['최대 심박수'].mean())
heartDiseaseData.head()

식별자  나이      성별    dataset            가슴 통증     혈압  콜레스테롤  공복 혈당  \
0    1  63    Male  Cleveland   typical angina  145.0  233.0   True   
1    2  67    Male  Cleveland     asymptomatic  160.0  286.0  False   
2    3  67    Male  Cleveland     asymptomatic  120.0  229.0  False   
3    4  37    Male  Cleveland      non-anginal  130.0  250.0  False   
4    5  41  Female  Cleveland  atypical angina  130.0  204.0  False   

              심전도  최대 심박수    협심증  ST 하강       ST 기울기  주요 혈관 수  \
0  lv hypertrophy   150.0  False    2.3  downsloping      0.0   
1  lv hypertrophy   108.0   True    1.5         flat      3.0   
2  lv hypertrophy   129.0   True    2.6         flat      2.0   
3          normal   187.0  False    3.5  downsloping      0.0   
4  lv hypertrophy   172.0  False    1.4    upsloping      0.0   

               탈라세미아  심장병 진단  
0       fixed defect       0  
1             normal       2  
2  reversable defect       1  
3             normal       0  
4             normal       0

#### 8-3. 협심증 칼럼에서 널값을 다 'False'로 번경하시오.

In [ ]:
heartDiseaseData['협심증'] = heartDiseaseData['협심증'].fillna(False)
heartDiseaseData['협심증'].head(5)

/var/folders/s0/flnd6hf15vs62n5vkdlhyg_m0000gn/T/ipykernel_93651/936540867.py:1: FutureWarning: Downcasting object dtype arrays on .fillna, .ffill, .bfill is deprecated and will change in a future version. Call result.infer_objects(copy=False) instead. To opt-in to the future behavior, set `pd.set_option('future.no_silent_downcasting', True)`
  heartDiseaseData['협심증'] = heartDiseaseData['협심증'].fillna(False)


0      False
1       True
2       True
3      False
4      False
       ...  
915    False
916    False
917    False
918    False
919     True
Name: 협심증, Length: 920, dtype: bool

### 9. 람다식을 이용하여 혈압을 다음 표를 기준으로 나눠 새로운 칼럼을 생성하시오.

| 분류 | 수축기 혈압 기준 |
| :--- | :--- |
| **저혈압** | $90 \text{ mmHg}$ 미만 |
| **정상 혈압** | $90 \text{ mmHg}$ ~ $119 \text{ mmHg}$ |
| **주의 혈압** | $120 \text{ mmHg}$ ~ $129 \text{ mmHg}$ |
| **고혈압 1기** | $130 \text{ mmHg}$ ~ $139 \text{ mmHg}$ |
| **고혈압 2기** | $140 \text{ mmHg}$ 이상 |

In [27]:
def classify_blood_pressure(bp):
    if bp < 90:
        return '저혈압'
    elif 90 <= bp <= 119:
        return '정상 혈압'
    elif 120 <= bp <= 129:
        return '주의 혈압'
    elif 130 <= bp <= 139:
        return '고혈압 1기'
    else :
        return '고혈압 2기'

heartDiseaseData['혈압 분류'] = heartDiseaseData['혈압'].apply(classify_blood_pressure)
heartDiseaseData.head()

식별자  나이      성별    dataset            가슴 통증     혈압  콜레스테롤  공복 혈당  \
0    1  63    Male  Cleveland   typical angina  145.0  233.0   True   
1    2  67    Male  Cleveland     asymptomatic  160.0  286.0  False   
2    3  67    Male  Cleveland     asymptomatic  120.0  229.0  False   
3    4  37    Male  Cleveland      non-anginal  130.0  250.0  False   
4    5  41  Female  Cleveland  atypical angina  130.0  204.0  False   

              심전도  최대 심박수    협심증  ST 하강       ST 기울기  주요 혈관 수  \
0  lv hypertrophy   150.0  False    2.3  downsloping      0.0   
1  lv hypertrophy   108.0   True    1.5         flat      3.0   
2  lv hypertrophy   129.0   True    2.6         flat      2.0   
3          normal   187.0  False    3.5  downsloping      0.0   
4  lv hypertrophy   172.0  False    1.4    upsloping      0.0   

               탈라세미아  심장병 진단   혈압 분류  
0       fixed defect       0  고혈압 2기  
1             normal       2  고혈압 2기  
2  reversable defect       1   주의 혈압  
3             normal       0  고혈압 1기  
4             normal       0  고혈압 1기